jsonpb/jsonpb.go

```diff
@@ -919,11 +919,12 @@ func (u *Unmarshaler) unmarshalValue(target reflect.Value, inputValue json.RawMe
 		if err := json.Unmarshal(inputValue, &slc); err != nil {
 			return err
 		}
-		len := len(slc)
-		target.Set(reflect.MakeSlice(targetType, len, len))
-		for i := 0; i < len; i++ {
-			if err := u.unmarshalValue(target.Index(i), slc[i], prop); err != nil {
-				return err
+		if l := len(slc); l > 0 {
```

Comment:
I think if user explicitly specifies an empty JSON list/map then the deserialized Go struct field should also be an empty slice/map.  If JSON field is not specified or if explicitly set to "null", then leave Go field as nil.